src/utils/route-handlers-middleware/middlewares/grpc-cluster-methods.ts

```diff
@@ -0,0 +1,41 @@
+import { isObjectLike } from 'lodash';
+
+import decodeUrlParams from '@/utils/decode-url-params';
+import {
+  type GRPCClusterMethods,
+  getClusterMethods,
+} from '@/utils/grpc/grpc-client';
+import { type GRPCMetadata } from '@/utils/grpc/grpc-service';
+
+import { type MiddlewareFunction } from '../route-handlers-middleware.types';
+
+const keyValuesAreStrings = (v: any): v is Record<string, string> => {
+  return (
+    isObjectLike(v) &&
+    Object.entries(v).every(
+      ([key, value]) => typeof key === 'string' && typeof value === 'string'
+    )
+  );
+};
+
+const grpcClusterMethods: MiddlewareFunction<
```

Comment:
if we assume the usage naming matches the export, then adding grpc would help putting context on what methods stands for

Reply:
In that case, I would suggest calling it `getGrpcClusterMethods` to differentiate it from the ctx property (and so that it's clear that it is a function)